conans/client/build/compilers_info.py

```diff
@@ -0,0 +1,282 @@
+
+
+def cppstd_flag(compiler, compiler_version, cppstd):
+    if not compiler or not compiler_version or not cppstd:
+        return ""
+    func = {"gcc": _cppstd_gcc,
+            "clang": _cppstd_clang,
+            "apple-clang": _cppstd_apple_clang,
+            "Visual Studio": _cppstd_visualstudio}.get(str(compiler), None)
+    flag = None
+    if func:
+        flag = func(str(compiler_version), str(cppstd))
+    return flag
+
+
+def cstd_flag(compiler, compiler_version, cstd):
+    if not compiler or not compiler_version or not cstd:
+        return ""
+    func = {"gcc": _cstd_gcc,
+            "clang": _cstd_clang,
+            "apple-clang": _cstd_apple_clang,
+            "Visual Studio": _cstd_visual_studio}.get(str(compiler), None)
+    flag = None
+    if func:
+        flag = func(compiler_version, cstd)
+    return flag
+
+
+def cppstd_default(compiler, compiler_version):
+
+    default = {"gcc": _gcc_cppstd_default(compiler_version),
+               "clang": "98",
+               "apple-clang": "98"}.get(str(compiler), None)
+    return default
+
+
+def cstd_default(compiler, compiler_version):
+
+    default = {"gcc": _gcc_cstd_default(compiler, compiler_version),
+               "clang": "11gnu",
+               "apple-clang": "11gnu"}.get(str(compiler), None)
+    return default
+
+
+def _gcc_cppstd_default(compiler_version):
+    if int(float(compiler_version)) <= 5:
+        return "98gnu"
+    if int(float(compiler_version)) <= 7:
+        return "14gnu"
+    return None
+
+
+def _gcc_cstd_default(compiler_version):
+    if int(float(compiler_version)) <= 4:
+        return "90gnu"
+    return "11gnu"
+
+
+def _cstd_visual_studio(_, __):
+    return None
+
+
+def _cppstd_visualstudio(visual_version, cppstd):
+
+    v14 = None
+    v17 = None
+
+    if int(visual_version) >= 14:
+        v14 = "c++14"
+        v17 = "c++latest"
+    if int(visual_version) >= 15:
+        v17 = "c++17"
+
+    flag = {"14": v14, "17": v17}.get(str(cppstd), None)
+    return "/std:%s" % flag if flag else None
+
+
+def _cstd_apple_clang(clang_version, cstd):
+    """https://github.com/Kitware/CMake/blob/1fe2dc5ef2a1f262b125a2ba6a85f624ce150dd2/
+    Modules/Compiler/GNU-CXX.cmake"""
+    v90 = v90gnu = v99 = v99gnu = v11 = v11gnu = None
+
+    if not float(clang_version) < 4.0:
+        v90 = "c90"
+        v90gnu = "gnu90"
+        v99 = "c99"
+        v99gnu = "gnu99"
+        v11 = "c1x"
+        v11gnu = "gnu1x"
+
+    flag = {"90": v90, "90gnu": v90gnu,
+            "99": v99, "99gnu": v99gnu,
+            "11": v11, "11gnu": v11gnu}.get(str(cstd), None)
+
+    return "-std=%s" % flag if flag else None
+
+
+def _cppstd_apple_clang(clang_version, cppstd):
+    """
+    Inspired in:
+    https://github.com/Kitware/CMake/blob/1fe2dc5ef2a1f262b125a2ba6a85f624ce150dd2/
+    Modules/Compiler/AppleClang-CXX.cmake
+    """
+
+    v98 = v98gnu = v11 = v11gnu = v14 = v14gnu = v17 = v17gnu = None
+
+    if not float(clang_version) < 4.0:
```

Comment:
Feels a bit hacky this comparison, not saying it doesnt work...

Reply:
yeah, suggestions?